app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -213,123 +281,118 @@ class SnapKeyringImpl implements SnapKeyringCallbacks {
-            );
-            await showSuccess(
-              this.#messenger,
-              snapId,
-              {
-                icon: IconName.UserCircleAdd,
-                title: t('snapAccountCreated'),
-              },
-              {
-                message: t('snapAccountCreatedDescription') as string,
-                address,
-                learnMoreLink,
-              },
-            );
-            // User has clicked on "OK"
-            trackSnapAccountEvent(
-              MetaMetricsEventName.AddSnapAccountSuccessClicked,
-            );
-          }
-
-          trackSnapAccountEvent(MetaMetricsEventName.AccountAdded);
-        } catch (e) {
-          // Error occurred while naming the account
-          const error = (e as Error).message;
-
-          await showError(
+        if (!skipConfirmation) {
+          // TODO: Add events tracking to the dialog itself, so that events are more
+          // "linked" to UI actions
+          // User should now see the "Successfuly added account" page
+          trackSnapAccountEvent(
+            MetaMetricsEventName.AddSnapAccountSuccessViewed,
+          );
+          await showSuccess(
             this.#messenger,
             snapId,
             {
               icon: IconName.UserCircleAdd,
-              title: t('snapAccountCreationFailed'),
+              title: t('snapAccountCreated'),
             },
             {
-              message: t(
-                'snapAccountCreationFailedDescription',
-                snapName,
-              ) as string,
+              message: t('snapAccountCreatedDescription') as string,
+              address,
               learnMoreLink,
-              error,
             },
           );
-
-          throw new Error(
-            `Error occurred while creating snap account: ${error}`,
+          // User has clicked on "OK"
+          trackSnapAccountEvent(
+            MetaMetricsEventName.AddSnapAccountSuccessClicked,
           );
         }
-      } else {
-        // User has cancelled account creation so remove the account from the keyring
-        await handleUserInput(accountNameConfirmationResult?.success);
 
-        throw new Error('User denied account creation');
+        trackSnapAccountEvent(MetaMetricsEventName.AccountAdded);
+      } catch (e) {
+        // Error occurred while naming the account
+        const error = (e as Error).message;
+
+        await showError(
+          this.#messenger,
+          snapId,
+          {
+            icon: IconName.UserCircleAdd,
+            title: t('snapAccountCreationFailed'),
+          },
+          {
+            message: t(
+              'snapAccountCreationFailedDescription',
+              snapName,
+            ) as string,
+            learnMoreLink,
+            error,
+          },
+        );
+
+        // FIXME: Should we still throw here? This part of the flow is not awaited, so that
+        // cannot be caught...
```

Comment:
Maybe just log it?

Reply:
Definitely, I'll do that, thanks!

Reply:
[9508ec2](https://github.com/MetaMask/metamask-extension/pull/30406/commits/9508ec2f6eb67ccf0c40e958e1feb5db5efe5e32)